src/main/java/com/minecolonies/coremod/commands/generalcommands/ScanCommand.java

```diff
@@ -0,0 +1,388 @@
+        {
+            player.displayClientMessage(Component.translatable(MAX_SCHEMATIC_SIZE_REACHED, Structurize.getConfig().getServer().schematicBlockLimit.get()), false);
+            return;
+        }
+
+        final long currentMillis = System.currentTimeMillis();
+        final String currentMillisString = Long.toString(currentMillis);
+        String fileName;
+        if (slot.getName().isEmpty())
+        {
+            fileName = Component.translatable("item.sceptersteel.scanformat", "", currentMillisString).getString();
+        }
+        else
+        {
+            fileName = slot.getName();
+        }
+
+        if (!fileName.contains(".blueprint"))
+        {
+            fileName+= ".blueprint";
+        }
+
+        final String[] split = fileName.split("/");
+        final String style = split.length <= 1 ? "" : split[0];
+
+        final BlockPos zero = new BlockPos(box.minX(), box.minY(), box.minZ());
+        final Blueprint
+          bp = BlueprintUtil.createBlueprint(world, zero, saveEntities, (short) box.getXSpan(), (short) box.getYSpan(), (short) box.getZSpan(), fileName, slot.getBox().getAnchor());
+
+        if (slot.getBox().getAnchor().isEmpty() && bp.getPrimaryBlockOffset().equals(new BlockPos(bp.getSizeX() / 2, 0, bp.getSizeZ() / 2)))
+        {
+            final List<BlockInfo> list = bp.getBlockInfoAsList().stream()
+                                           .filter(blockInfo -> blockInfo.hasTileEntityData() && blockInfo.getTileEntityData().contains(TAG_BLUEPRINTDATA))
+                                           .collect(Collectors.toList());
+
+            if (list.size() > 1)
+            {
+                player.displayClientMessage(Component.translatable("com.ldtteam.structurize.gui.scantool.scanbadanchor", fileName), false);
+            }
+        }
+
+        Network.getNetwork().sendToPlayer(new SaveScanMessage(BlueprintUtil.writeBlueprintToNBT(bp), fileName), (ServerPlayer) player);
+        if (style.isEmpty())
+        {
+            return;
+        }
+
+        boolean hasJigsaw = false;
+        for (BlockState state : bp.getPalette())
+        {
+            if (state.getBlock() == Blocks.JIGSAW)
+            {
+                hasJigsaw = true;
+                break;
+            }
+        }
+
+        if (!hasJigsaw)
+        {
+            return;
+        }
+
+        final BlockState primary = world.getBlockState(zero.offset(bp.getPrimaryBlockOffset()));
+        final boolean isHut;
+        if (primary.getBlock() instanceof AbstractBlockHut<?>)
+        {
+            isHut = true;
+            final AbstractTileEntityColonyBuilding building = (AbstractTileEntityColonyBuilding) world.getBlockEntity(zero.offset(bp.getPrimaryBlockOffset()));
+            building.addTag(new BlockPos(0, 0, 0), "deactivated");
+            building.addTag(new BlockPos(0, 0, 0), style);
```

Comment:
`TileEntityColonyBuilding.reactivate()` assumes that the style tag is the first non-`deactivated` tag.  However this code will add it to the end, which means it won't work if the hut block has some other tags already...

It also appears to expect a more complicated tag value than just `style` -- it wants `Long Pack Name:filename` (unless it's a built-in pack, then you can cheat a bit).

This whole thing seems a bit fragile.  Perhaps there's a better way to store it?  Is there some reason it doesn't use the existing hut NBT properties?

Reply:
I'll look again into it when I come home. I'll make sure to add it to the front.